experimental/dds/tree2/src/shared-tree/sharedTree.ts

```diff
@@ -241,3 +241,15 @@ export class SharedTreeFactory implements IChannelFactory {
 		return tree;
 	}
 }
+
+/**
+ * SharedTree class object containing static factory method used for {@link @fluidframework/fluid-static#IFluidContainer}.
+ *
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class SharedTreeFactoryObject {
```

Comment:
This being available here still doesn't solve your use case (or others with the same problem); it doesn't get you the class object you need to satisfy the container create API. For that, you also need the `SharedTreeFactoryObject as unknown as SharedObjectClass<ISharedTree>` cast. If the goal of putting it here is to add this "kludge" to a central location so that others can use the kludge as well, then it should probably be the "full kludge". As it is now, a consumer could get this class, but then they still have to do the cast; at that point, their hands are dirty enough that it wouldn't have been much more work for them to also add these four lines of boilerplate.

I think the most helpful thing would be to put all the kludginess in one place and have it well documented. I would suggest:

* Make this class generic over the type of factory, and have it take a factory when constructed, which is the factory returned by `getFactory()`.
* Don't expose it. Instead, expose a free function which takes a factory, creates an instance of this class, and returns a `SharedObjectClass` (it does the cast on the return value).
* Move the function and class to the fluidstatic package and import from there.
* Document the free function thoroughly as a workaround, add a TODO, etc.

That way, the kludge is well-bounded and communicated. 

Or, maybe we can just change fluid static to take a factory instead of SharedObjectClass?

Reply:
@ssimic2 @ChumpChief Interested in your thoughts here as it pertains to the fluid-static API. In particular, would it make sense to update fluid-static to take in a factory rather than `SharedObjectClass`?

Reply:
[This comment](https://github.com/microsoft/FluidFramework/pull/15813#discussion_r1214862980) has some additional context that we've discussed with @Abe27342 as well, namely a potential additional motivation for updating the fluid-static APIs going forward.